ddtrace/appsec/_iast/_taint_tracking/__init__.py

```diff
@@ -95,17 +116,29 @@ def taint_pyobject(pyobject: Any, source_name: Any, source_value: Any, source_or
     if source_origin is None:
         source_origin = OriginType.PARAMETER
 
-    pyobject_newid = set_ranges_from_values(pyobject, len(pyobject), source_name, source_value, source_origin)
+    try:
+        pyobject_newid = set_ranges_from_values(pyobject, len(pyobject), source_name, source_value, source_origin)
+    except ValueError:
+        iast_taint_log_error("Failed to taint pyobject, taint_pyobject")
```

Comment:
A little redundant (taint pyobject, taint_pyobject). Also we could print the pyobject on the error, but I am not sure if that could have PII implications.